src/coreclr/nativeaot/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.llvm.xml

```diff
@@ -0,0 +1,22 @@
+<linker>
+  <assembly fullname="System.Private.CoreLib">
+    <type fullname="System.Numerics.Vector">
+      <method signature="System.Boolean get_IsHardwareAccelerated()" body="stub" value="false" />
+    </type>
+    <type fullname="System.Environment">
+      <method signature="System.Int32 get_ProcessorCount()" body="stub" value="1" />
+    </type>
+    <type fullname="System.Runtime.CompilerServices.RuntimeFeature">
+      <method signature="System.Boolean get_IsDynamicCodeCompiled()" body="stub" value="false" />
+    </type>
+    <type fullname="System.Resources.ResourceReader">
+      <method signature="System.Boolean InitializeBinaryFormatter()" body="stub" value="false" />
+    </type>
+    <type fullname="System.Globalization.GlobalizationMode">
+      <method signature="System.Boolean get_UseNls()" body="stub" value="false" />
```

Comment:
Is this going unconditionally trim the full globalization support? Do we even need `System.Globalization.Native` in that case?

Reply:
Well if I remove the library from the linking, then I get back
```
EXEC : warning : undefined symbol: GlobalizationNative_ChangeCase (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_ChangeCaseInvariant (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_ChangeCaseTurkish (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_CloseSortHandle (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_CompareString (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_EndsWith (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetCalendarInfo (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetCalendars (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetDefaultLocaleName (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetJapaneseEraStartDate (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetLatestJapaneseEra (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetLocaleInfoGroupingSizes (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetLocaleInfoInt (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetLocaleInfoString (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetLocaleName (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetLocaleTimeFormat (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetSortHandle (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_IndexOf (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_InitOrdinalCasingPage (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_LastIndexOf (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_StartsWith (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
```
So it seems there are still some things left in.  `GlobalizationNative_StartsWith` for example is rooted up through a few calls to `S_P_CoreLib_System_Globalization_CompareInfo__IcuStartsWith` and then `string.StartsWith`  I could go through the code seeing if there's some other substitution that should be added?

Reply:
Ah, I have confused Nls and Icu. This makes sense.